code/game/objects/effects/proximity.dm

```diff
@@ -35,65 +30,51 @@
 	QDEL_LAZYLIST(checkers)
 	return ..()
 
-/datum/proximity_monitor/proc/HandleMove()
+/datum/proximity_monitor/proc/HandleMove(atom/movable/mover, atom/old_loc)
 	SIGNAL_HANDLER
 
-	var/atom/_host = host
-	var/atom/new_host_loc = _host.loc
+	var/atom/new_host_loc = host.loc
 	if(last_host_loc != new_host_loc)
 		last_host_loc = new_host_loc //hopefully this won't cause GC issues with containers
-		var/curr_range = current_range
-		SetRange(curr_range, TRUE)
-		if(curr_range)
-			testing("HasProx: [host] -> [host]")
-			hasprox_receiver.HasProximity(host) //if we are processing, we're guaranteed to be a movable
-
+		SetRange(current_range, TRUE)
+
+/**
+ * Sets the detection range.
+ *
+ * Arguments:
+ * * range - how many tiles around host should we detect. 0 means only my own tile, negative values turn the monitor off
+ * * force_rebuild - rebuild the checkers even when the range is the same as before
+ */
 /datum/proximity_monitor/proc/SetRange(range, force_rebuild = FALSE)
 	if(!force_rebuild && range == current_range)
-		return FALSE
-	. = TRUE
+		return
 
 	current_range = range
+	var/old_checkers_len = LAZYLEN(checkers)
 
-	var/list/checkers_local = checkers
-	var/old_checkers_len = checkers_local.len
-
-	var/atom/_host = host
-
-	var/atom/loc_to_use = ignore_if_not_on_turf ? _host.loc : get_turf(_host)
-	if(wire && !isturf(loc_to_use)) //it makes assemblies attached on wires work
-		loc_to_use = get_turf(loc_to_use)
-	if(!isturf(loc_to_use)) //only check the host's loc
-		if(range)
-			var/obj/effect/abstract/proximity_checker/pc
-			if(old_checkers_len)
-				pc = checkers_local[old_checkers_len]
-				--checkers_local.len
-				QDEL_LAZYLIST(checkers_local)
-			else
-				pc = new(loc_to_use, src)
-
-			checkers_local += pc //only check the host's loc
+	var/atom/loc_to_use = host.loc
+	if(!isturf(loc_to_use) || range < 0)
+		QDEL_LAZYLIST(checkers)
 		return
 
-	var/list/turfs = RANGE_TURFS(range, loc_to_use)
+	var/list/turfs = RANGE_TURFS(range, host.loc)
```

Comment:
I'm curious, why change this line?

Reply:
i've removed loc_to_use var altogether in the first version, but then i partially reversed it

Reply:
Ah, right. No need for it then.
```suggestion
	var/list/turfs = RANGE_TURFS(range, loc_to_use)
```